src/Orleans/Providers/ILeaseProvider.cs

```diff
@@ -81,6 +107,23 @@ public class LeaseRequest
         /// Duration of the lease
         /// </summary>
         public TimeSpan Duration { get; set; }
```

Comment:
This is only immutable if their is no 'set' on any of the properties after construction.